api/upload.go

```
@@ -225,3 +307,63 @@ func (c *Client) UploadSimpleWithRate(url string, metadata []byte, rate float64,
 func (c *Client) UploadSimple(url string, metadata []byte, files ...*UploadSource) (chan int64, chan error) {
 	return c.UploadSimpleWithRate(url, metadata, 0, files...)
 }
+
+func (c *Client) UploadSimpleSignedUrl(url string, metadata []byte, files ...*UploadSource) (chan int64, chan error) {
+	var progress chan int64
+	var errChan chan error
+	var intermediateErrChan chan error
+	var completeTicket bool
+	var err error
+
+	ticket, _, _ := c.CreateUploadTicket(url, metadata, files...)
```

We should distinguish between the case where signed urls are not supported, and a server error occurs. I would also consider adding retry logic (with larger timeouts) to the CreateUploadTicket and CompleteUploadTicket calls, if SDK is now taking ownership of the retry functionality.

---

With a server error, would we want to return the error or try the normal upload through the api?

---

I'm not sure I understand the need for retry loops for the Create and Complete Ticket methods, we have the retry loop to guard against the 500s from google, but I wouldn't expect any transient errors.

---

The primary situation is if we're in the middle of a maintenance window while jobs are completing, and engine cannot talk to core.